### ISSUE TITLE
Bun.serve - Request-Body size increase without reading

### ISSUE THREAD

**Issue by niklas-stegmann** [oven-sh/bun#12263]
### What version of Bun is running?

1.1.17+bb66bba1b

### What platform is your computer?

Microsoft Windows NT 10.0.19045.0 x64

### What steps can reproduce the bug?

1. Setting up a simple http server with bun
```ts
Bun.serve({
  async fetch(req) {
    function sleep(ms: number): Promise<void> {
      return new Promise((resolve) => setTimeout(resolve, ms));
    }

    console.log(req.body);

    await sleep(1000 * 60 * 60);

    return new Response("Bun!");
  },
  port: 3000,
  maxRequestBodySize: Number.MAX_SAFE_INTEGER
});
```

2. Send a large file
`curl --data-binary "@./1GB.bin" http://localhost:3000`

### What is the expected behavior?

As with NodeJS, the server does not use much RAM, because nothing is read from the body and saved in a variable.

![image](https://github.com/oven-sh/bun/assets/32418474/f6112670-8203-4ccf-b463-a729916c7b14)

```js
var http = require("http");

http
  .createServer(async function (req, res) {
    function sleep(ms) {
      return new Promise((resolve) => setTimeout(resolve, ms));
    }

    req.on("readable", function () {
      try {
        req.read();
      } catch (error) {}
    });
    req.on("end", function () {
      console.log("finished");
    });

    await sleep(1000 * 60 * 60);

    res.end();
  })
  .listen(3000);
```

### What do you see instead?

The RAM of the server has increased by the size of the uploaded file.

![image](https://github.com/oven-sh/bun/assets/32418474/32c9e15c-073d-48ef-bf45-5487de23074e)

**Comment by Jarred-Sumner** [comment]
This happens due to the use of `sleep` (or any long-running async task).  We are reading data from the socket so that the next request can be processed and so that the developer is able to read the request body after an async task has completed.

We could tweak the behavior here though and make it only start reading in the data once the getReader() or equivalent function has been called.

**Comment by niklas-stegmann** [comment]
Sounds like a good adjustment. My initial problem in my project was that I couldn't prevent the entire file / request body from being read. I want to send the body directly to AWS S3 without consuming much memory in my application.

Example:

```javascript
const command = new PutObjectCommand({
  Bucket: "test-bucket",
  Key: "test.bin",
  Body: Readable.fromWeb(body)
});

await s3Client.send(command);
```

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.1.22
